src/components/AttachmentPicker/index.native.js

```diff
@@ -177,30 +178,36 @@ class AttachmentPicker extends Component {
       * Common image picker handling
       *
       * @param {function} imagePickerFunc - RNImagePicker.launchCamera or RNImagePicker.launchImageLibrary
+      * @param {Boolean} isCamera
       * @returns {Promise<ImagePickerResponse>}
       */
-    showImagePicker(imagePickerFunc) {
+    showImagePicker(imagePickerFunc, isCamera = false) {
         return new Promise((resolve, reject) => {
-            imagePickerFunc(getImagePickerOptions(this.props.type), (response) => {
-                if (response.didCancel) {
-                    // When the user cancelled resolve with no attachment
-                    return resolve();
-                }
-                if (response.errorCode) {
-                    switch (response.errorCode) {
-                        case 'permission':
-                            this.showPermissionsAlert();
-                            break;
-                        default:
-                            this.showGeneralAlert();
-                            break;
-                    }
-
-                    return reject(new Error(`Error during attachment selection: ${response.errorMessage}`));
-                }
+            if (!isCamera) {
+                this.openImagePicker(imagePickerFunc, resolve, reject);
+            } else {
+                checkCameraPermission().then((isValid) => {
+                    isValid ? this.openImagePicker(imagePickerFunc, resolve, reject) : this.showPermissionsAlert();
+                });
+            }
+        });
+    }
 
-                return resolve(response.assets);
-            });
+    openImagePicker(imagePickerFunc, resolve, reject) {
+        imagePickerFunc(getImagePickerOptions(this.props.type), (response) => {
+            if (response.didCancel) {
+                // When the user cancelled resolve with no attachment
+                return resolve();
+            }
+            if (response.errorCode) {
```

Comment:
Any reason we changed switch case to if-else here?

Reply:
Resolved.